qt/widgets/common/src/TableWidgetValidators.cpp

```diff
@@ -0,0 +1,84 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidQtWidgets/Common/TableWidgetValidators.h"
+
+namespace {
+
+std::string OR(const std::string &lhs, const std::string &rhs) { return "(" + lhs + "|" + rhs + ")"; }
+std::string NATURAL_NUMBER_WITH_PRECISION(std::size_t digits) {
+  return OR("0", "[1-9][0-9]{," + std::to_string(digits - 1) + "}");
+}
+// non numeric characters
+const std::string EMPTY = "^$";
+const std::string SPACE = "(\\s)*";
+const std::string COMMA = SPACE + "," + SPACE;
+const std::string DASH = "\\-";
+
+// number and numeric sets
+const std::string NATURAL_NUMBER = "(0|[1-9][0-9]*)";
+const std::string REAL_NUMBER = "(-?" + NATURAL_NUMBER + "(\\.[0-9]*)?)";
+const std::string REAL_RANGE = "(" + REAL_NUMBER + COMMA + REAL_NUMBER + ")";
+const std::string NUMBER = NATURAL_NUMBER_WITH_PRECISION(4);
+const std::string NATURAL_RANGE = "(" + NUMBER + DASH + NUMBER + ")";
+const std::string NATURAL_OR_RANGE = OR(NATURAL_RANGE, NUMBER);
+
+// final lists
+const std::string MASK_LIST = "(" + REAL_RANGE + "(" + COMMA + REAL_RANGE + ")*" + ")|" + EMPTY;
+const std::string SPECTRA_LIST = "(" + NATURAL_OR_RANGE + "(" + COMMA + NATURAL_OR_RANGE + ")*)";
+
+constexpr auto NUMERICAL_PRECISION = 6;
+
+} // namespace
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+std::string getRegexValidatorString(const regexValidatorStrings &validatorMask) {
+  switch (validatorMask) {
+  case regexValidatorStrings::SpectraValidator:
+    return SPECTRA_LIST;
+  case regexValidatorStrings::MaskValidator:
+    return MASK_LIST;
+  default:
+    return SPECTRA_LIST;
```

Comment:
For the default, we might want to throw a `std::logic_error` because it indicates  a programmatic error has been made by the programmer if an enum which isn't `SPECTRA_LIST` or `MASK_LIST` has been provided. Or maybe they have added a new enum but have forgot to update this part of the code. An example is seen here

https://github.com/mantidproject/mantid/blob/2eb0cae186fedca519c3bfe539017fb7140eff82/Framework/API/src/Algorithm.cpp#L360